### ISSUE TITLE
What is ng in build command

### ISSUE THREAD

**Issue by danionescu2007** [kaltura/kmc-ng#309]
I am following the guide in the readme to set-up the kmc-ng on a ubuntu 16 xenial. I really don't know if this is the optimal setup, but I managed to get it running behind a nginx proxy.
I reached the point where is says to run `$ ng build --prod`.
Firstly I could not figure it out what the ng binary was about. I found one matching this name in `~/kmc-ng/node_modules/.bin`. After running the command with "build --prod" params, I get an error in red color in the terminal:
```
ERROR in Error encountered resolving symbol values statically. Calling function 'KalturaLogger', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol ContentBulkLogAppComponent in /root/kmc-ng/src/applications/content-bulk-log-app/content-bulk-log-app.component.ts, resolving symbol ContentBulkLogAppComponent in /root/kmc-ng/src/applications/content-bulk-log-app/content-bulk-log-app.component.ts

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/root/kmc-ng/src'
 @ ./src/main.ts 5:0-74
 @ multi ./src/main.ts
```

The full output with the command is:
```
root@kaltura:~/kmc-ng/node_modules/.bin# ./ng build --prod
Hash: 2fc059facabbf80c5206
Time: 20298ms
chunk    {0} polyfills.7341fa6f91204ec0f3da.bundle.js (polyfills) 510 kB {4} [initial] [rendered]
chunk    {1} main.ee0a0b91f6babfd19ba7.bundle.js (main) 12.7 kB {3} [initial] [rendered]
chunk    {2} styles.71bc48c40682dd4768df.bundle.css (styles) 69 bytes {4} [initial] [rendered]
chunk    {3} vendor.33a2c1968dd8a235b223.bundle.js (vendor) 852 kB [initial] [rendered]
chunk    {4} inline.04d093d3b4e320d4c07c.bundle.js (inline) 0 bytes [entry] [rendered]

ERROR in Error encountered resolving symbol values statically. Calling function 'KalturaLogger', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol ContentBulkLogAppComponent in /root/kmc-ng/src/applications/content-bulk-log-app/content-bulk-log-app.component.ts, resolving symbol ContentBulkLogAppComponent in /root/kmc-ng/src/applications/content-bulk-log-app/content-bulk-log-app.component.ts

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/root/kmc-ng/src'
 @ ./src/main.ts 5:0-74
 @ multi ./src/main.ts
root@kaltura:~/kmc-ng/node_modules/.bin# cd ..
root@kaltura:~/kmc-ng/node_modules# cd ..
root@kaltura:~/kmc-ng# node_modules/.bin/ng
ng        ngc       ng-xi18n
root@kaltura:~/kmc-ng# node_modules/.bin/ng build --prod
Hash: 2fc059facabbf80c5206
Time: 18989ms
chunk    {0} polyfills.7341fa6f91204ec0f3da.bundle.js (polyfills) 510 kB {4} [initial] [rendered]
chunk    {1} main.ee0a0b91f6babfd19ba7.bundle.js (main) 12.7 kB {3} [initial] [rendered]
chunk    {2} styles.71bc48c40682dd4768df.bundle.css (styles) 69 bytes {4} [initial] [rendered]
chunk    {3} vendor.33a2c1968dd8a235b223.bundle.js (vendor) 852 kB [initial] [rendered]
chunk    {4} inline.04d093d3b4e320d4c07c.bundle.js (inline) 0 bytes [entry] [rendered]

ERROR in Error encountered resolving symbol values statically. Calling function 'KalturaLogger', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol ContentBulkLogAppComponent in /root/kmc-ng/src/applications/content-bulk-log-app/content-bulk-log-app.component.ts, resolving symbol ContentBulkLogAppComponent in /root/kmc-ng/src/applications/content-bulk-log-app/content-bulk-log-app.component.ts

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/root/kmc-ng/src'
 @ ./src/main.ts 5:0-74
 @ multi ./src/main.ts
```

Any help will be much appreciated

**Comment by kaltura-hooks** [comment]
Hi @danionescu2007,

Thank for you reporting an issue and helping improve Kaltura!

To get the fastest response time, and help the maintainers review and test your reported issues or suggestions, please ensure that your issue includes the following (please comment with more info if you have not included all this info in your original issue):
 - Is the issue you're experiencing consistent and across platforms? or does it only happens on certain conditions?
    please provide as much details as possible.
 - Which Kaltura deployment you're using: Kaltura SaaS, or self-hosted?
	If self hosted, are you using the RPM or deb install?
 - Packages installed.
	When using RPM, paste the output for:
```
	# rpm -qa \"kaltura*\"
```	
	For deb based systems:
```	
	# dpkg -l \"kaltura-*\"
```
 - If running a self hosted ENV - provide the MySQL server version used
 - If running a self hosted ENV - is this a single all in 1 server or a cluster?
 - If running a self hosted ENV, while making the problematic request, run:
```
	# tail -f /opt/kaltura/log/*.log /opt/kaltura/log/batch/*.log | grep -A 1 -B 1 --color \"ERR:\|PHP\|trace\|CRIT\|\[error\]\"
```
   and paste the output.
 - When relevant, provide any screenshots or screen recordings showing the issue you're experiencing.

For general troubleshooting see:
https://github.com/kaltura/platform-install-packages/blob/Jupiter-10.13.0/doc/kaltura-packages-faq.md#troubleshooting-help

If you only have a general question rather than a bug report, please close this issue and post at:
http://forum.kaltura.org

Thank you in advance,

**Comment by esakal** [comment]
Hello @danionescu2007,
We are going to deploy next version during the first week of January 2018. Please try then and let us know if it still happens.

Sincerely,
Eran

**Comment by esakal** [comment]
@danionescu2007,
We published version 3.4.1 which seems to build correctly with production flag. 

Can you please retry and let us know the status.

Thanks,
Eran.

**Comment by esakal** [comment]
I'm closing this issue for housekeeping. If this error still happens please open a new issue.
